site/src/docs/foundation/guidelines/server-side-rendering.mdx

```diff
@@ -0,0 +1,171 @@
+---
+slug: "/foundation/guidelines/server-side-rendering"
+title: "Server-side rendering"
+navTitle: 'Server-side rendering'
+---
+import { Accordion, Notification } from 'hds-react';
+import LeadParagraph from '../../../components/LeadParagraph';
+
+# Server-side rendering
+
+<LeadParagraph>
+  The server-side rendering support of the Helsinki Design System makes the user's landing experience to pages smooth without
+  flashes of unstyled content.
+</LeadParagraph>
+
+## What is server-side rendering?
+
+Server-side rendering (SSR) is the process where the rendering of HTML pages is done on the server-side. The fully rendered HTML
+document is then sent to the browser. The rendering of the HTML can happen at build time (Static site generation
+or pre-rendering), or during an HTTP request. The alternative to SSR is client-side rendering (CSR), where most of the HTML
+content is composed and rendered in the browser using JavaScript.
+
+## How does HDS support server-side rendering?
+
+For HDS components to work with server-side rendering, in server you need to inject the critical CSS styles
+of the HDS components that are being used into the initially rendered HTML. For hds-core, you must include the styles from
+provided CSS files yourself. For hds-react, HDS provides multiple options, that we will cover next.
+
+<Notification label="Critical CSS" className="siteNotification">
+Critical CSS is the CSS applied to above-the-fold elements. It is the CSS of the immediately visible content in the browser
+viewport when the user opens your website. Critical CSS does not usually have the styles of the elements that are in the
+scrollable content outside of the browser viewport. There is one exception though, and that is if you have anchor links on the page.
+When user opens website from an url that includes an anchor link, the browser automatically scrolls the page so that the link
+will be visible.
+</Notification>
+
+### Option 1: getCriticalHdsRules tool to get critical styles (recommended)
+
+HDS React components use CSS-in-JS in a way where the styles are by default injected into the head tag in the browser.
+If you use server-side rendering and do not include the critical styles on the server, this leads to flashes of unstyled
+content when the user lands on the page.
+
+To include critical styles in server, HDS exposes a tool for extracting the used critical styles of HDS components.
```

Comment:
"**on the** server". Google says that it's ok to use either in or on prepositions with server but there seems to be some consensus among programmers that on is a bit more logical.

Reply:
- [ ] <b>on the</b> server